server/controllers/notification.js

```diff
@@ -0,0 +1,97 @@
+const asyncHandler = require("express-async-handler");
+const Notification = require("../models/Notification");
+const User = require("../models/User");
+
+exports.createNotification = asyncHandler(async (req, res) => {
+  const { title, type, senderId, recipientId, message } = req.body;
+  await User.findOne({ _id: senderId }, function (err) {
+    if (err) {
+      res.status(404).send("Sender not found!!");
+    }
+  });
+  await User.findOne({ _id: recipientId }, function (err) {
+    if (err) {
+      res.status(404).send("Receiver not found");
+    }
+  });
+
+  try {
+    const notification = await Notification.create({
+      type,
+      title,
+      recipientId,
+      senderId,
+      message,
+      date,
+    });
+    res.status(201).json(notification);
+  } catch (err) {
+    res.status(500).json(err);
+  }
+});
+
+exports.getAllNotifications = asyncHandler(async (req, res) => {
+  const { recipientId } = req.user.id;
+  await User.findOne({ _id: recipientId }, function (err) {
+    if (err) {
+      res.status(404).send("Receiver not found!");
+    }
+  });
+  try {
+    const allNotifications = await Notification.find({
+      _id: recipientId,
```

Comment:
Won't we need to check the receipientId field in the notification document and not the actual document's id?

Eg: `const allNotifications = await Notification.find({ recipientId: recipientId)..`

Reply:
that's right, I have corrected the code